tests/qunit/tests/js/api.util.js

```diff
@@ -899,18 +900,134 @@ require(['gh.core', 'gh.api.tests'], function(gh, testAPI) {
+                                             * @private
+                                             */
+                                            var _validateParent = function(_orgUnit) {
+                                                if (_orgUnit.ParentId && _orgUnit.ParentId !== _orgUnit.Parent.id) {
+                                                    assert.fail('The organisational unit\'s parent object doesn\'t correspond with the parent ID');
+                                                }
+
+                                                if (_orgUnit.Parent) {
+                                                    _validateParent(_orgUnit.Parent);
+                                                }
+                                            };
+
+                                            // Start validating the tree
+                                            _validateParent(serie.borrowedFrom);
+                                        }
+                                    }
+                                });
+                            }
+
+                            QUnit.start();
+                        });
+                    });
+
+                } else {
+                    QUnit.start();
+                }
+            });
+        });
+    });
+
+    // Test the 'addParentInfoToOrgUnit' functionality
+    QUnit.asyncTest('addParentInfoToOrgUnit', function(assert) {
+        expect(1);
+
+        // Retrieve the tripos structure for the test app
+        var testApp = testAPI.getTestApp();
+        gh.utils.getTriposStructure(testApp.id, function(err, data) {
+            assert.ok(!err);
+
+            // Verify that all the parts that have a parent are decorated with their parent
+            _.each(data.parts, function(part) {
+                if (part.ParentId && part.ParentId !== part.Parent.id) {
+                    assert.fail('Verify that the organisational unit is decorated with its correct parent');
+                }
+            });
+            QUnit.start();
+        });
+    });
+
+    QUnit.asyncTest('getTriposStructure', function(assert) {
+        expect(3);
+
+        // Verify that an error is thrown when an invalid value for app id was provided
+        assert.throws(function() {
+            gh.utils.getTriposStructure('invalid_app_id', function() {});
+        }, 'Verify that an error is thrown when an invalid value for app id was provided');
+
         // Verify that an error is thrown when no callback was provided
         assert.throws(function() {
-            gh.utils.getTriposStructure();
+            gh.utils.getTriposStructure(null);
         }, 'Verify that an error is thrown when no callback was provided');
 
         // Verify that an error is thrown when an invalid callback was provided
         assert.throws(function() {
-            gh.utils.getTriposStructure('invalid_callback');
+            gh.utils.getTriposStructure(null, 'invalid_callback');
```

Comment:
I placed a comment above asking to test with an empty appId but looks like you're already doing this. Can you just go ahead and remove the `istanbul ignore` from that line in the utility file?